### PR TITLE
Message Index sorting by invalid columns

### LINKED ISSUE #4075: Non-default sorting on middle pages
Sometimes when sorting the message index by specific columns and you have several pages + aren't on the first or last page, it'll try to `order by` invalid column names.

> http://example.com/index.php?board=14.440;sort=last_post
> Unknown column 'tid_last_msg' in 'order clause'
> File: Sources/MessageIndex.php Line: 305

It applies the `identifier` sanitization method to functions as well:
> http://example.com/index.php?board=14.440;sort=starter
> Unknown column 'COALESCE(memfreal_name, mfposter_name)' in 'order clause'
> File: Sources/MessageIndex.php Line: 305

### PR DESCRIPTION
**Pull request by DuncanFairley**
Fixes #4075
The values in `$_REQUEST['sort']` are already pulled from `$sort_methods` so they're already safe.

### CODE REVIEW

File: Sources/Subs-Db-mysql.php [hunk 1]
```diff
@@ -272,7 +272,7 @@ function smf_db_replacement__callback($matches)
 
 		case 'identifier':
 			// Backticks inside identifiers are supported as of MySQL 4.1. We don't need them for SMF.
-			return '`' . strtr($replacement, array('`' => '', '.' => '')) . '`';
```
Comment (colinschoen) [inline Sources/Subs-Db-mysql.php:275]: Why is this change needed now if you are using `raw`?

**Review by colinschoen — CHANGES_REQUESTED** [review]

File: Sources/Subs-Db-mysql.php (hunk 1 above)
Comment (DuncanFairley) [reply Sources/Subs-Db-mysql.php:275]: I feel that the `itentifier` filter should either accept table.field syntax, or return an error - instead of just mangling the input. A couple sanitization libraries I've used seem to accept table.field.

Comment (colinschoen) [reply Sources/Subs-Db-mysql.php:275]: Agreed.

**Review by colinschoen — APPROVED** [review]
LGTM approval

**Comment by albertlast** [comment]
@colinschoen this pr is in my eyes wrong in many direction
it's do only mysql stuff (pg is effected to),
with the replace of the point with the point the sql issue keep there.

**Comment by colinschoen** [comment]
I'll take a closer look tomorrow this does need to be fixed for PG but the solution presented works for MySQL.

**Comment by albertlast** [comment]
#4077 fix this issue correctly for both databases

**Comment by DuncanFairley** [comment]
In #4077 you drop the aliased table name which would break other sort columns from `$sort_methods` like `mf.subject`, as there's multiple `messages` tables joint you'd get an ambiguous error.

**Comment by albertlast** [comment]
first important to know,
your solution works never because you had the table alias and the col nam in the same quote.
see http://sqlfiddle.com/#!9/582dd8/2

the order by work that the col name of the select statement part is taken
and there can only be exists one messages all other needs to be a different name.

**Comment by DuncanFairley** [comment]
But my change doesn't only put quotes around the whole thing... That was the point of my change. 
http://sandbox.onlinephpfunctions.com/code/be39d01ea1ee2214075bf7d7bbabf9a06e03f0d0
> Outputs \`table\`.\`column\`

**Comment by albertlast** [comment]
And why did you change the type of the sql statement from identifier to raw?

**Comment by albertlast** [comment]
okay gotcha because of the coalesec
